modules/minimongo-observable/observable-cursor.ts

```diff
@@ -79,13 +80,31 @@ export class ObservableCursor<T> extends Observable<T[]> {
     });
   }
 
+  addedAt(doc, at, before) {
+    this.data.splice(at, 0, doc);
+    this.handleChange();
+  }
+
+  changedAt(doc, old, at) {
+    this.data[at] = doc;
```

Comment:
Not sure actually but I noticed this https://github.com/meteor/meteor/issues/5678 issue so I though it might be an idea to remove fetch. The `foreach` in `fetch` does quite a bit underneath actually. If the query result contains only a subset of the full subscription it's also possible that the cursor will have walk several steps in the minimongo storage. There's also the problem for the cursor to step through sorted data.

Now, I honestly don't know for sure that it's an issue and I haven't had the time to sit down and figure out a way to actually benchmark this.